go/chat/search/indexer.go

```diff
@@ -477,6 +480,31 @@ func (idx *Indexer) allConvs(ctx context.Context, uid gregor1.UID) (map[string]t
 	return convMap, nil
 }
 
+func (idx *Indexer) flattenConvMap(ctx context.Context, uid gregor1.UID,
```

Comment:
is this necessary if the results are ordered by message ctime afterwards?

Reply:
I think it's better like this, since normally recent chats do have what you are looking for, and so the amount of re-ordering on the frontend is less. If we find it is slow I can take it out.